### PR TITLE
#176: add decimal format cast in Hive

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java [hunk 1]
```diff
@@ -266,4 +266,19 @@ protected void checkImportPropertyConsistency(final String importFromProperty, f
...
+    protected void validateCastNumberToDecimalProperty(final String castNumberToDecimalProperty)
+            throws PropertyValidationException {
+        if (this.properties.containsKey(castNumberToDecimalProperty)) {
+            final Pattern pattern = Pattern.compile("\\s*(\\d+)\\s*,\\s*(\\d+)\\s*");
+            final String precisionAndScale = this.properties.get(castNumberToDecimalProperty);
+            final Matcher matcher = pattern.matcher(precisionAndScale);
+            if (!matcher.matches()) {
+                throw new PropertyValidationException("Unable to parse adapter property " + castNumberToDecimalProperty
+                        + " value \"" + precisionAndScale
+                        + " into a number precison and scale. The required format is \"<precsion>.<scale>\", where "
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]:
```suggestion
                        + " into a number's precision and scale. The required format is \"<precision>.<scale>\", where "
```

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import static com.exasol.adapter.dialects.hive.HiveProperties.HIVE_CAST_NUMBER_TO_DECIMAL_PROPERTY;
+
+/**
+ * This class implements Hive-specific reading of column metadata.
+ */
+public class HiveColumnMetadataReader extends BaseColumnMetadataReader {
+
+    /**
+     * Create a new instance of the {@link HiveColumnMetadataReader}
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java]:
```suggestion
     * Create a new instance of the {@link HiveColumnMetadataReader}.
```

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public HiveColumnMetadataReader(final Connection connection, final AdapterProperties properties,
+            final IdentifierConverter identifierConverter) {
+        super(connection, properties, identifierConverter);
+    }
+
+    @Override
+    public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
+        switch (jdbcTypeDescription.getJdbcType()) {
+        case Types.DECIMAL:
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java]: Replace with `if/else` unless we have more cases.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            return mapDecimal(jdbcTypeDescription);
+        default:
+            return super.mapJdbcType(jdbcTypeDescription);
+        }
+    }
+
+    protected DataType mapDecimal(final JdbcTypeDescription jdbcTypeDescription) {
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java]: This is duplication with the `OracleColumnMetadataReader`. Please move the common part to `BaseColumnMetadataReader` as `protected` methods.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveProperties.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.exasol.adapter.dialects.hive;
+
+/**
+ * This class contains hive-specific adapter properties.
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveProperties.java]:
```suggestion
 * This class contains HIVE-specific adapter properties.
```

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReaderTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import com.exasol.adapter.dialects.*;
+import com.exasol.adapter.jdbc.*;
+import com.exasol.adapter.metadata.*;
+
+class HiveColumnMetadataReaderTest {
+    private HiveColumnMetadataReader columnMetadataReader;
+
+    @BeforeEach
+    void beforeEach() {
+        this.columnMetadataReader = new HiveColumnMetadataReader(null, AdapterProperties.emptyProperties(),
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReaderTest.java]: Move to `AbstractColumnMetadataReaderTest` (also from the Oracle counterpart) to reduce duplication.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java]: Duplication removed

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReaderTest.java (hunk 6 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReaderTest.java]: done

**Pull request merged**